basic-app/components/wallet/RegisterWallet.tsx

```diff
@@ -16,28 +17,46 @@ gql`
     }
   }
 `
+interface Props {
+  blockchain: Blockchain
+}
 
-export function RegisterWallet() {
+export function RegisterWallet({ blockchain }: Props) {
   const flowUser = useFlowUser()
+  const metamask = useMetaMask()
 
   const { mutate: registerWallet, error, isLoading } = useWalletMutation(useRegisterWalletMutation)
 
   // When the user logs in, register their wallet
   useEffect(() => {
-    if (!flowUser?.addr) {
+    if (blockchain !== Blockchain.Flow || !flowUser?.addr) {
       return
     }
 
     registerWallet({ address: flowUser.addr })
-  }, [flowUser?.addr, flowUser?.loggedIn, registerWallet])
+  }, [blockchain, flowUser?.addr, flowUser?.loggedIn, registerWallet])
```

Comment:
This effect is flow-blockchain only?

Reply:
yea looks like flow blockchain doesnt give you a promise when you connect to it so its needed for that only